src/rgw/rgw_data_sync.cc

```diff
@@ -2657,6 +2689,17 @@ int RGWBucketShardIncrementalSyncCR::operate()
       }
     } while (!list_result.empty() && sync_status == 0);
 
+    if (syncstopped) {
+      drain_all();
+
+      yield {
+        const string& oid = RGWBucketSyncStatusManager::status_oid(sync_env->source_zone, bs);
+        RGWRados *store = sync_env->store;
+        call(new RGWRadosRemoveCR(store, rgw_raw_obj{store->get_zone_params().log_pool, oid}));
```

Comment:
directly after this `[delete]` op succeeds, i see a `[call lock.unlock]` op fail with `-ENOENT`. can you try doing a `lease_cr->abort()` here to avoid sending that unlock?